File: backend/prostitute-protection/src/main/java/de/eshg/prostituteprotection/domain/model/Facility.java
/*
 * Copyright 2025 cronn GmbH
 * SPDX-License-Identifier: Apache-2.0
 */

package de.eshg.prostituteprotection.domain.model;

import de.eshg.lib.procedure.domain.model.RelatedFacility;
import jakarta.persistence.Entity;
import jakarta.persistence.Index;
import jakarta.persistence.Table;

@Entity
@Table(indexes = @Index(columnList = "procedure_id"))
public class Facility extends RelatedFacility<ProstituteProtectionProcedure> {}

File: packages/lib-portal/src/formatters/facility.ts
/**
 * Copyright 2026 cronn GmbH
 * SPDX-License-Identifier: Apache-2.0
 */

import { isDefined, isNullish } from "remeda";

import { Nullable } from "../types/utility";

interface Facility {
  name: string;
}

export function formatFacilityName(facility: Nullable<Partial<Facility>>) {
  if (isNullish(facility) || !isDefined(facility.name)) {
    return "";
  }

  return facility.name;
}
